packages/app/src/app/overmind/namespaces/dashboard/actions.ts

```diff
@@ -1978,3 +1979,40 @@ export const updateAlbum = async (
     effects.notificationToast.error('There was a problem updating album');
   }
 };
+
+export const getContributionBranches = async ({ state, effects }: Context) =>
+  // eslint-disable-next-line consistent-return
+  {
+    const { dashboard } = state;
+    try {
+      let contributionBranches: NonNullable<
+        ContributionBranchesQuery['me']
+      >['recentBranches'][number][] = [];
+      dashboard.contributions = null;
+
+      const contributionsData = await effects.gql.queries.getContributionBranches(
+        {}
+      );
+      if (
+        !contributionsData ||
+        !contributionsData.me ||
+        !contributionsData.me.recentBranches
+      ) {
+        return;
+      }
+      contributionBranches = contributionsData.me.recentBranches;
+
+      if (!contributionBranches) {
+        return;
+      }
```

Comment:
We already check for the `contributionsData.me.recentBranches` in the if statement above.

I (nitpick) personally like a more "positive" approach where we check for certain values instead of doing early returns. Unless it causes multiple nested `if` statements. But right now I think it would work:

```ts
if (contributionsData?.me?.recentBranches) {
  contributionBranches = contributionsData.me.recentBranches;

  dashboard.contributions = contributionBranches.map(branch => ({
    type: 'contribution-branch',
    branch,
  }));
}
```

Reply:
@tristandubbeld Updated the action in commit: [dc65d5a](https://github.com/codesandbox/codesandbox-client/pull/6906/commits/dc65d5a820413f44ac7fb4b3a7052e0caaeac003). Personally, I prefer early returns and the other actions are following the same pattern. I don't think we have any guidelines on this yet but might be a topic for the next Huddle.js.